tests/scripts/thread-cert/otci/otci/otci.py

```diff
@@ -0,0 +1,1871 @@
+            self.execute_command(f'dataset masterkey {master_key}')
+
+        if network_name is not None:
+            self.execute_command(f'dataset networkname {self.__escape_escapable(network_name)}')
+
+        if panid is not None:
+            self.execute_command(f'dataset panid {panid}')
+
+        if pskc is not None:
+            self.execute_command(f'dataset pskc {pskc}')
+
+        if security_policy is not None:
+            rotation_time, flags = security_policy
+            self.execute_command(f'dataset securitypolicy {rotation_time} {flags}')
+
+        if pending_timestamp is not None:
+            self.execute_command(f'dataset pendingtimestamp {pending_timestamp}')
+
+    # TODO: dataset mgmtgetcommand
+    # TODO: dataset mgmtsetcommand
+    # TODO: dataset set <active|pending> <dataset>
+
+    #
+    # Allowlist management
+    #
+
+    def enable_allowlist(self):
+        self.execute_command(f'macfilter addr {self.__detect_allowlist_cmd()}')
+
+    def disable_allowlist(self):
+        self.execute_command('macfilter addr disable')
+
+    def add_allowlist(self, addr: str, rssi: int = None):
+        cmd = f'macfilter addr add {addr}'
+
+        if rssi is not None:
+            cmd += f' {rssi}'
+
+        self.execute_command(cmd)
+
+    def remove_allowlist(self, addr: str):
+        self.execute_command(f'macfilter addr remove {addr}')
+
+    def clear_allowlist(self):
+        self.execute_command('macfilter addr clear')
+
+    def set_allowlist(self, allowlist: Collection[Union[str, Tuple[str, int]]]):
+        self.clear_allowlist()
+
+        if allowlist is None:
+            self.disable_allowlist()
+        else:
+            self.enable_allowlist()
+            for item in allowlist:
+                if isinstance(item, str):
+                    self.add_allowlist(item)
+                else:
+                    addr, rssi = item[0], item[1]
+                    self.add_allowlist(addr, rssi)
+
+    # TODO: denylist
+    # TODO: macfilter rss
+    # TODO: macfilter rss add <extaddr> <rss>
+    # TODO: macfilter rss add-lqi <extaddr> <lqi>
+    # TODO: macfilter rss remove <extaddr>
+    # TODO: macfilter rss clear
+
+    def __detect_allowlist_cmd(self):
+        if self.api_version >= 28:
+            return 'allowlist'
+        else:
+            return '\x77\x68\x69\x74\x65\x6c\x69\x73\x74'
+
+    #
+    # Unicast Addresses management
+    #
+    def add_ipaddr(self, ip: Union[str, ipaddress.IPv6Address]):
+        """Add an IPv6 address to the Thread interface."""
+        self.execute_command(f'ipaddr add {ip}')
+
+    def del_ipaddr(self, ip: Union[str, ipaddress.IPv6Address]):
+        """Delete an IPv6 address from the Thread interface."""
+        self.execute_command(f'ipaddr del {ip}')
+
+    def get_ipaddrs(self) -> Tuple[Ip6Addr]:
+        """Get all IPv6 addresses assigned to the Thread interface."""
+        return tuple(map(Ip6Addr, self.execute_command('ipaddr')))
```

Comment:
Ip addresses and prefixes are not using symbol "::" for shortened addresses to its maximum capability.
```
e.g.
# ot-ctl ipaddr add 2001::1
Done
# ot-ctl ipaddr
2001:0:0:0:0:0:0:1
fe80:0:0:0:509d:4d64:74a0:9c53
Done

Similar issue is seen with prefixes also.
# ot-ctl prefix add cafe:beef:0::/64 paros med
Done
# ot-ctl prefix
fd00:abba:cafe:0::/64 aos med fffe
fd00:0:0:0::/64 aos med fffe
cafe:beef:0:0::/64 paros med fffe
Done
```

Reply:
prefix compressed in OTCI.